### PR TITLE
Drawer widget styles

### PR DESCRIPTION
**Pull request by TeyaVes**
Spec: https://github.com/telerik/kendo/issues/9519
Design: https://github.com/telerik/kendo-ux-private/issues/20

LESS PR: https://github.com/telerik/kendo/pull/9618

### CODE REVIEW

File: packages/default/scss/drawer/_layout.scss [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+                z-index: 10002;
+            }
+
+            .k-drawer-items {
+                max-width: 80vw;
+            }
+        }
+
+
+        &.push {
```
Comment (PreslavKozovski) [inline packages/default/scss/drawer/_layout.scss]: use the "k-" prefix

File: packages/default/scss/drawer/_layout.scss [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+            }
+        }
+
+        .k-overlay {
+            display: none;
+        }
+
+        &.expanded  {
+            .k-overlay {
+                display: block;
```
Comment (PreslavKozovski) [inline packages/default/scss/drawer/_layout.scss]: as far as I see, this selector will never work, and this style is assigned via JS. Thus, I'd suggest removing it

File: packages/default/scss/drawer/_layout.scss [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+            }
+        }
+
+        &.right {
```
Comment (PreslavKozovski) [inline packages/default/scss/drawer/_layout.scss]: use the "k-" prefix

File: packages/default/scss/drawer/_layout.scss [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+            &.overlay {
```
Comment (PreslavKozovski) [inline packages/default/scss/drawer/_layout.scss]: use the "k-" prefix

File: packages/default/scss/drawer/_layout.scss [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+                .k-drawer-container {
+                    left: auto;
+                    right: 0;
+                }
+            }
+
+            &.push {
```
Comment (PreslavKozovski) [inline packages/default/scss/drawer/_layout.scss]: use the "k-" prefix

File: packages/default/scss/drawer/_layout.scss [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+                .k-drawer-container {
+                    order: 2;
+                }
+            }
+
+            .k-drawer-container.expanded,
+            &.mini-mode .k-drawer-container {
```
Comment (PreslavKozovski) [inline packages/default/scss/drawer/_layout.scss]: use the "k-" prefix

File: packages/default/scss/drawer/_layout.scss [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+                }
+            }
+
+            &.push {
+                .k-drawer-container {
+                    order: 2;
+                }
+            }
+
+            .k-drawer-container.expanded,
```
Comment (PreslavKozovski) [inline packages/default/scss/drawer/_layout.scss]: use the "k-" prefix

File: packages/default/scss/drawer/_layout.scss [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+                border-right-width: 0;
+            }
+        }
+    }
+
+    .k-rtl,
+    [dir="rtl"] {
+
+        .k-drawer-wrapper {
+            &.push {
```
Comment (PreslavKozovski) [inline packages/default/scss/drawer/_layout.scss]: use the "k-" prefix

Also, refactor the selectors to 

`&.left.push {`

File: packages/default/scss/drawer/_layout.scss [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+            .k-drawer-container.expanded,
+            &.mini-mode .k-drawer-container {
+                border-left-width: $drawer-border-width;
+                border-right-width: 0;
+            }
+        }
+    }
+
+    .k-rtl,
+    [dir="rtl"] {
```
Comment (PreslavKozovski) [inline packages/default/scss/drawer/_layout.scss]: This selector will not work if the `dir="rtl"` attribute is assigned to the `k-drawer` element;

The selector should look like:

```
.k-rtl .k-drawer,
.k-drawer[dir="rtl"],
[dir="rtl"] .k-drawer{
}


Also, the below code is not handling the `overlay` scenario; are we going to support this?
```

File: packages/default/scss/drawer/_layout.scss [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+
+        .k-drawer-wrapper {
+            &.push {
+                .k-drawer-container {
+                    order: 2;
+                }
+            }
+
+            &.right {
```
Comment (PreslavKozovski) [inline packages/default/scss/drawer/_layout.scss]: use the "k-" prefix

File: packages/default/scss/drawer/_layout.scss [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+                &.push {
```
Comment (PreslavKozovski) [inline packages/default/scss/drawer/_layout.scss]: use the "k-" prefix

Also, refactor the selectors to 

`&.right.push {`

File: packages/default/scss/drawer/_layout.scss [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+                    .k-drawer-container {
+                        order: initial;
```
Comment (PreslavKozovski) [inline packages/default/scss/drawer/_layout.scss]: This is not working in IE

File: packages/default/scss/drawer/_layout.scss [hunk 13]
```diff
@@ -0,0 +1,168 @@
+@include exports("drawer/layout") {
+
+    .k-drawer-container,
+    .k-drawer-container .k-drawer-items {
```
Comment (PreslavKozovski) [inline packages/default/scss/drawer/_layout.scss]: These styles can go in the next section with the following selector:

```
&,
.k-drawer-items {
....
```

File: packages/default/scss/drawer/_layout.scss [hunk 14]
```diff
@@ -0,0 +1,168 @@
...
+        border-style: solid;
+        border-width: 0;
+        box-sizing: border-box;
+        overflow-x: hidden;
+        white-space: nowrap;
+
+        .k-drawer-items {
+            padding: 0;
+            width: 0;
+            transition: all .3s ease-out;
```
Comment (PreslavKozovski) [inline packages/default/scss/drawer/_layout.scss]: for consistency with the other Kendo styles, use `300ms`

File: tests/visual/drawer.html [hunk 15]
```diff
@@ -0,0 +1,330 @@
...
+                    <div class="k-drawer-content" style="padding: 0 20px;">Lorem ipsum dolor sit amet consectetur adipisicing elit.olor sit amet consectetur
+                            adipisicing elit. Error accusantium odit, optio nulla maiores quo neque fugit debitis dignissimos incidunt
+                            maxime? Eum voluptatem blanditiis voluptatum praesentium dolorem, dolore placeat debitis quod delectus laborum
+                            assumenda cupiditate.
+                    </div>
+                </div>
+            </div>
+        </section>
+
+        <section class="k-rtl">
```
Comment (PreslavKozovski) [inline tests/visual/drawer.html]: Add another HTML file for the RTL tests

File: packages/default/scss/drawer/_theme.scss [hunk 16]
```diff
@@ -0,0 +1,62 @@
+@include exports("drawer/theme") {
```
Comment (PreslavKozovski) [inline packages/default/scss/drawer/_theme.scss]: Add an empty row

File: packages/default/scss/drawer/_theme.scss [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+    .k-drawer-container {
+        background-color: $drawer-bg;
+        color: $drawer-text;
+        border-color: $drawer-border;
+    }
+
+    .k-drawer-container,
+    .k-drawer-container .k-drawer-items {
```
Comment (PreslavKozovski) [inline packages/default/scss/drawer/_theme.scss]: If you decide to move the relevant styles in the `_layout`, apply the same logic here.

File: packages/material/scss/_variables.scss [hunk 18]
```diff
@@ -1289,6 +1289,42 @@ $adaptive-scheduler-current-text: $accent;
...
+
+$drawer-hovered-text: $hovered-text !default;
+$drawer-hovered-bg: $hovered-bg !default;
+
+$drawer-focused-bg: $hovered-bg !default;
+$drawer-focused-shadow: none !default;
+
+$drawer-selected-text: $secondary !default;
+$drawer-selected-bg: $base-bg !default;
+
```
Comment (PreslavKozovski) [inline packages/material/scss/_variables.scss]: For consistency, remove this empty row or add it in the other `_variables.scss` files

File: packages/default/scss/drawer/_layout.scss (hunk 9 above)
Comment (TeyaVes) [reply packages/default/scss/drawer/_layout.scss]: the overlay mode adds fixed position and left or right 0 depending on the position so we don't need additional styling in the RTL

**Pull request merged**